src/propslib/DelegationConfig.cpp

```diff
@@ -238,7 +245,10 @@ try
     {
         if (pkg.console.clsid == defCon && pkg.terminal.clsid == defTerm)
         {
-            chosenPackage = pkg;
+            if (!isTerminalAsDefault || pkg.console.allowDefaultTerminalHandoff)

```

Comment:
Ohhhhhhhhhhhhhhhhhhhhh okay I see now. The `CheckShouldTerminalBeDefault` blocks will set the respective `defCon` and `defTerm` to the clsid's for Terminal Stable. Okay. Now it all makes sense.

Reply:
I'm not quite sure how to best structure the code here. But it would probably flow better in general if `s_GetDefaultConsoleId` and `s_GetDefaultTerminalId` were one function. 🤔